extensions/vertx-http/deployment-spi/src/main/java/io/quarkus/vertx/http/deployment/spi/GeneratedStaticResourceBuildItem.java

```diff
@@ -28,8 +28,14 @@ public final class GeneratedStaticResourceBuildItem extends MultiBuildItem {
     private final byte[] content;
 
     private GeneratedStaticResourceBuildItem(final String endpoint, final byte[] content, final Path file) {
-        if (!requireNonNull(endpoint, "endpoint is required").startsWith("/")) {
-            throw new IllegalArgumentException("endpoint must start with '/'");
+        requireNonNull(endpoint, "endpoint is required");
+        if (!endpoint.startsWith("/")) {
+            throw new IllegalArgumentException(
+                    "GeneratedStaticResourceBuildItem endpoint must start with '/': %s".formatted(endpoint));
+        }
+        if (endpoint.endsWith("/")) {
+            throw new IllegalArgumentException(
+                    "GeneratedStaticResourceBuildItem endpoint must not end with '/': %s".formatted(endpoint));
```

Comment:
Hmmm, sorry, but I'm not very comfortable doing that.

I think you should tweak the condition as I was suggesting before.

My main reason is that people might have decided to generate a `GeneratedStaticResourceBuildItem` for the directory themselves to work around the issue and it might contain a `/`. If we want it to be backportable, we need to be less aggressive.

Does it make sense?

Reply:
Not really, it should have been checked in the first place. We are reading the file on disk, adding a directory does not make sense and would fail another way..

Reply:
I suppose we should even add a check that the file exists (and print an error if not) before reading it in the processor

Reply:
I'm specifically talking about other producers of `GeneratedStaticResourceBuildItem`.

A build item can be generated by any extension around. You can't be sure that all the extensions out there are not pushing something with `/`. Especially since they might do that to work around this very issue and get the directory referenced in the native image. And also because it's actually not that big of a deal to have an ending `/`, except for the newly added code, or am I missing something?

We can't break compatibility in back branches as easily as that.

Now if you absolutely want to check this in main, but I don't really understand the value of being so strict tbh so maybe you could explain. We will need two patches, one for main, and one for the back branches.

Reply:
I pasted the code of what is done with this BuildItem, we read the file. It won't work if it ends with `/` whether after or before this PR (this code is just to ensure we have the proper error and check as soon as possible).

https://github.com/quarkusio/quarkus/pull/49455/files#diff-7bd378a321691086398f7630d2e5cf0a810695723812e4d2dbffe959201e200aR56-R70